src/jupyter_contrib_nbextensions/nbextensions/execution_dependencies/execution_dependencies.js

```diff
@@ -0,0 +1,96 @@
+define([
+    'base/js/namespace',
+    'notebook/js/codecell'
+], function (
+    Jupyter,
+    codecell
+) {
+    "use strict";
+
+    var CodeCell = codecell.CodeCell;
+
+    return {
+        load_ipython_extension: function () {
+            console.log('[exec_deps] patching CodeCell.execute');
+            var orig_execute = codecell.CodeCell.prototype.execute;                                                 // get original cell execute function
+            CodeCell.prototype.execute = function (stop_on_error) {
+                var root_tags = this.metadata.tags || [];
+                if(root_tags != [] && root_tags.some(tag => /=>.*/.test(tag))) {                                    // if the root cell contains any dependencies, resolve dependency tree...
+                    var root_cell = this;
+                    var identified_cells = Jupyter.notebook.get_cells().filter(function (cell, idx, cells) {        // ...get all cells which have at least one id (these are the only ones we could have in deps)
+                        var tags = cell.metadata.tags || [];                        
+                        return (cell === root_cell || tags.some(tag => /#.*/.test(tag)));
+                    });
+
+                    console.log('Collect ids and dependencies...');
+                    var cell_map = {}  
+                    var dep_graph = {}
+                    identified_cells.forEach(function (cell) {                                                      // ...get all identified cells (the ones that have at least one #tag)
+                        var tags = cell.metadata.tags || [];
+                        var identities = tags.filter(tag => /#.*/.test(tag)).map(tag => tag.substring(1));          // ...get all identities and drop the #
+                        if(cell === root_cell && !tags.some(tag => /#.*/.test(tag))) {
+                            identities.push("DD27AE1D138027D0D7AB824FD0DDDC61367D5CCA4AAB42CE50840762B053764D");    // ...generate an id for the root cell for internal usage
+                        }
+
+                        var deps = tags.filter(tag => /=>.*/.test(tag)).map(tag => tag.substring(2));               // ...get all dependencies and drop the =>
+                        identities.forEach(function (id) {
+                            cell_map[id] = cell;
```

Comment:
this line implicitly means only a single cell can have each `#some_id` tag. Is this what you want? It could make sense to have a few cells with the same tag, so do something like 

```javascript
cell_map[id] = cell_map[id] || [];
cell_map[id].push(cell);
```

Alternatively, if you don't intend to support this, at least let the user know that adding multiple cells with the same identifier tag won't work correctly in the readme.

Reply:
>  Is the cell_id you mentioned above unique within the notebook's context? Then I could use that for uniqueness and allow multiple cells to have the same hashtag id.

Yes, it should be unique for the current session. See [notebook/static/notebook/js/cell.js#L101](https://github.com/jupyter/notebook/blob/3de5e501f3f1eb16d89830b658d03b9c3e1a4c8b/notebook/static/notebook/js/cell.js#L101) for where it gets assigned, and [notebook/static/base/js/utils.js#L206-L220](https://github.com/jupyter/notebook/blob/3de5e501f3f1eb16d89830b658d03b9c3e1a4c8b/notebook/static/base/js/utils.js#L206-L220) for the implementation

Reply:
Sounds both reasonable. I will look into this as soon as the rest works.